packages/x-components/src/x-modules/next-queries/store/types.ts

```diff
@@ -70,6 +78,18 @@ export interface NextQueriesMutations extends StatusMutations, QueryMutations {
    * @param params - The new extra params.
    */
   setParams(params: Dictionary<unknown>): void;
+
+  /**
+   * Adds a new entry to the result's dictionary.
+   *
+   * @param payload - Object containing the next query, the totalResults and the results to add.
+   */
+  setResults(payload: SetNextQueryPreviewResultsPayload): void;
```

Comment:
Can we simplify the type to be like this? makes sense?

```suggestion
  setResults(payload: Dictionary<NextQueryPreviewResults>): void;
```